### PR TITLE
Completed the form for adding assignment.

### PR DESCRIPTION
**Pull request by m-e-l-u-h-a-n**
Note:
It presently contains a bug which is with editing tasks and review criterions .
It can be fixed easily upon api integration. For now after editing the criterion you need to explicitly save them.

### CODE REVIEW

**Comment by king-11** [comment]
@m-e-l-u-h-a-n perform an interactive rebase and remove the not needed commits and merges while check again why the build is failing

File: layouts/createassignmentlayout.vue [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+      </v-container>
+    </v-main>
+    <v-footer
+      :absolute="!fixed"
+      app
+    >
+      <span>&copy; {{ new Date().getFullYear() }}</span>
+    </v-footer>
+  </v-app>
+</template>
```
Comment (king-11) [inline layouts/createassignmentlayout.vue:20]: Remove the layout use default instead

File: nuxt.config.js [hunk 2]
```diff
@@ -81,14 +82,12 @@ export default {
...
-    }
-  },
-  /*
-  ** Build configuration
-  ** See https://nuxtjs.org/api/configuration-build/
-  */
-  build: {
-  }
-}
+    },
```
Comment (king-11) [inline nuxt.config.js:87]: build error could have been caused here check in this file

File: .gitignore [hunk 3]
```diff
@@ -117,3 +117,6 @@ sw.*
 
 # TernJS port file
 .tern-port
+
+# local vsCode settings
+.vscode/
```
Comment (king-11) [inline .gitignore:122]: Cool

**Review by king-11 — CHANGES_REQUESTED** [review]

**Comment by m-e-l-u-h-a-n** [comment]
> @m-e-l-u-h-a-n perform an interactive rebase and remove the not needed commits and merges while checking again why the build is failing by doing
> 
> ```shell
> git rebase -i HEAD~10
> yarn build
> ```
Build may be falling due to possible change in config file when i was trying to resolve some merge conflict yesterday.

**Review by MadhavaDasari — CHANGES_REQUESTED** [review]
Looks like a grate work 👌

- Looks like there is a some bug in the evaluating the weightage and total no of points ( it's always showing the error snackbar ) 

- i think it better to keep the save button active , so that user knows the required feilds  when he clicks it (due to error messages  ) , other wise he will not be knowing the what are the required feilds ( just after seeing the form ) .. or else think of an other alternative

File: pages/createassignment.vue [hunk 4]
```diff
@@ -0,0 +1,287 @@
...
+          <v-card-text justify="center">
+            <v-form ref="form" v-model="valid" @submit="saveAssignment">
+              <v-text-field
+                label="Title"
+                counter
+                maxlength="100"
+                v-model="title"
+                :rules="[rules.minLength, rules.titleLength]"
+                placeholder="Name of the assignment"
+              />
```
[the remark this answers is not in the document]
Comment (m-e-l-u-h-a-n) [reply pages/createassignment.vue]: would do those changes during api integration

File: pages/createassignment.vue [hunk 5]
```diff
@@ -0,0 +1,287 @@
...
+          return false;
+      }
+      return false;
+    },
+    // validation of assignment
+    validateAssignment() {
+      return (
+        this.title.length > 0 &&
+        this.submissionDeadline >= new Date().toISOString() &&
+        this.reviewDeadline > this.submissionDeadline
```
[the remark this answers is not in the document]
Comment (m-e-l-u-h-a-n) [reply pages/createassignment.vue]: seems quite resonable

**Pull request closed without merging**